prisma/migrations/20240109235224_auto_withdraw/migration.sql

```diff
@@ -0,0 +1,70 @@
+-- AlterTable
+ALTER TABLE "Withdrawl" ADD COLUMN     "autoWithdraw" BOOLEAN NOT NULL DEFAULT false;
+
+-- AlterTable
+ALTER TABLE "users" ADD COLUMN     "autoWithdrawMaxFeePercent" DOUBLE PRECISION,
+ADD COLUMN     "autoWithdrawThreshold" INTEGER,
+ADD COLUMN     "lnAddr" TEXT;
+
+CREATE OR REPLACE FUNCTION user_auto_withdraw() RETURNS TRIGGER AS $$
+DECLARE
+BEGIN
+    INSERT INTO pgboss.job (name, data)
+    SELECT 'autoWithdraw', jsonb_build_object('id', NEW.id)
+    -- only if there isn't already a pending job for this user
+    WHERE NOT EXISTS (
+        SELECT *
+        FROM pgboss.job
+        WHERE name = 'autoWithdraw'
+        AND data->>'id' = NEW.id::TEXT
+        AND state = 'created'
+    );
```

Comment:
You might be right, but I think this is clearer considering neither of us seems to know exactly how singletonKey works. As is, a future reader doesn't need to understand how pgboss works in detail.

Reply:
> considering neither of us seems to know exactly how singletonKey works

Valid point, we don't and it's probably not worth to change that - at least not just for this here.

Was just worried about bugs that we might have missed which a library that is specifically built for this did not miss. But the code seems to be simple enough that we can rely on it. Maybe - when I have time - I'll spend some time to look into the source code of `pgboss` to see how they are doing it and how close we are. Maybe there is something we can learn.

> As is, a future reader doesn't need to understand how pgboss works in detail.

Mhh, makes sense, even though I found it funny first because at my previous company, database triggers were only used as a last resort - also because of future readers who might not know much about the database schema and thus might run into unexpected problems because they aren't aware of the triggers - or that triggers even can exist? I don't remember exactly.

But I guess it's different here since `pgboss` is a library and not something as fundamental as a database trigger.